parlai/clib/cuda/ngram_repeat_block_cuda.cpp

```diff
@@ -0,0 +1,50 @@
+/*
```

Comment:
Is our code freshly rewritten, or did we get it from fastseq. We need to maintain the original copyright headers (in addition to our own) if that's the case. Both are MIT so it's no problem, but we need to include Copyright (c) Microsoft etc

Reply:
Thanks for pointing this out! We definitely referenced their code but i've also made some significant changes. I think we should add the copyright here? can you let me know what's the right move? not super familiar with this.

Reply:
As we discussed in our mtg, you can cite the original file and say we adapt from it. And maintain the Microsoft copyright header.